### ISSUE TITLE
Missing characters in nano (and maybe some other tools)

### ISSUE THREAD

**Issue by mzellho** [cmderdev/cmder#1827]
### Purpose of the issue
- [X] Bug report (encountered problems/errors)
- [ ] Feature request (request for new functionality)
- [X] Question

### Description of the issue 
I noticed some strange behaviour with a fresh cmder installation (including git bash) when I connect to remote hosts via ssh and call up nano. Some files open up and miss their first character. If I type a character at the first position in the file, the missing character gets visible and the newly typed character is prepended to the formerly missing character.

Note: startup command is {cmd::Cmder as Admin}, the exact same things are working fine if startup command is {bash::mintty} or {bash::bash} or if I am connecting using putty or an out-of-the-box portable ConEmu.

Any ideas? Did I mess up something in my environment?

**Comment by namanyayg** [comment]
I see this as well. Cmder/Windows. Happens with other programs like `vi` as well. Any solutions?

**Comment by swnesbitt** [comment]
same here

**Comment by namanyayg** [comment]
Try doing `which ssh` to see what `ssh` cmder is using, it must be set to the one in the `<your cmder directory>\vendor\git-for-windows\bin` folder. If not, add an entry at the top of the PATH for this and ensure that it's using the right `ssh.exe` file. Mine was previously set to OpenSSH's client.

**Comment by mzellho** [comment]
Thanks, same here:

```
λ which ssh
/c/WINDOWS/System32/OpenSSH/ssh
```

To me this resolves this issue.

**Comment by dtbarne** [comment]
To be clear `/c/WINDOWS/System32/OpenSSH/ssh` is the ssh client causing the issue. Change that to a better one, such as the git-for-windows one and cmder is usable again!

**Comment by OceansCrashing** [comment]
not sure if something changed, but the git-for-windows folder only contains bash.exe, git.exe, and sh.exe (not ssh.exe)  Strangely, it still worked when adding it to the path.  I instead added <cmder_dir>\vendor\git-for-windows\usr\bin which contains both sh.exe and ssh.exe and that works as well.  Anything wrong with doing this?  Out of curiosity, why does both work?

**Comment by adiov** [comment]
Apologies for poking at an old issue.

Using Git for Windows instead of the `System32/OpenSSH` version  _sorta_ worked, but it caused a somewhat unrelated issue for me (couldn't run more than one SSH session, but it's not relevant here). In the end, what worked for me was setting `TERM=xterm` as an environment variable.

![image](https://user-images.githubusercontent.com/7467822/54631974-b89e5200-4a85-11e9-8c67-e1ce6019cf12.png)

Now I'm able to use nano normally with `/c/WINDOWS/System32/OpenSSH/ssh.exe` without any issues.

**Comment by namanyayg** [comment]
@ThreepE0 You're right, there's no `ssh.exe` in the directory I mentioned and both seem to work. `/usr/bin` seems to have many other utilities though so I've edited my comment above to reflect that, thanks.

`nano` still seems to break a lot of things. To fix this, I learned `vi(m)` ;). Would be great if something got `nano` to work. 

@adiov fwiw, setting `TERM` to `xterm` causes `nano` to completely break and not show any output at all, at least with `ssh.exe` set to cmder's version. Also changes colors in SSH Vim so ended up setting it back to default (which seems to be `cygwin` when I do `echo %TERM`).
